src/services/rateSource/PrivatBankService.ts

```diff
@@ -0,0 +1,37 @@
+import { RateSourceResponse, RateSourceService } from './RateSourceService';
+import responseMessages from '../../constants/responseMessages';
+
+type PrivatBankResponse = {
+  ccy: 'EUR' | 'USD';
+  base_ccy: 'UAH';
+  buy: number;
+  sale: number;
+}[];
+
+const { SOMETHING_WENT_WRONG } = responseMessages;
+
+class PrivatBankService extends RateSourceService {
+  constructor() {
+    super('https://api.privatbank.ua/p24api/pubinfo');
+  }
+
+  async retrieve(): Promise<RateSourceResponse> {
+    const rawResponse = await fetch(this.url);
```

Comment:
Looks like you have a getter for URL getUrl, so no need to fetch it with this.url :)
Same comment to other services